mantaray/marshal.go

```diff
@@ -0,0 +1,140 @@
+package mantaray
+
+import (
+	"errors"
+	"fmt"
+)
+
+const (
+	preambleSize = 64
```

Comment:
Is the assumption that everything needs to be in the 64 bytes part (not asking just related to this value)? I ask because if the intention was to have smallest encoding* (*in some cases), we could always add size before actual array.

Reply:
i dont understand this